buffalo/cmd/generate/action.go

```diff
@@ -0,0 +1,141 @@
+// Copyright © 2016 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aymerick/raymond"
+	"github.com/bep/inflect"
+	"github.com/markbates/gentronics"
+	"github.com/spf13/cobra"
+)
+
+var runningTests = false
+
+//ActionCmd is the cmd that generates actions.
+var ActionCmd = &cobra.Command{
+	Use:     "action [name] [actionName...]",
+	Aliases: []string{"a"},
+	Short:   "Generates new action(s)",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 2 {
+			return errors.New("you should provide action name and handler name at least")
+		}
+
+		if _, err := os.Stat("actions"); err != nil {
+			return errors.New("actions directory not found, ensure you're inside your buffalo folder")
+		}
+
+		name := args[0]
+		actions := args[1:]
+
+		data := gentronics.Data{"under": inflect.Underscore(name)}
+		_, err := os.Stat(filepath.Join("actions", fmt.Sprintf("%v.go", data["under"])))
+		fileExists := err == nil
+
+		g := gentronics.New()
+
+		if !fileExists {
+			g.Add(gentronics.NewFile(filepath.Join("actions", fmt.Sprintf("%s.go", data["under"])), rActionFileT))
+			g.Add(gentronics.NewFile(filepath.Join("actions", fmt.Sprintf("%s_test.go", data["under"])), rActionTest))
+			if err = g.Run(".", data); err != nil {
+				return err
+			}
+		}
+
+		for _, action := range actions {
+			g.Add(buildActionAppender(name, action))
+		}
+
+		if !runningTests {
+			g.Add(Fmt)
+		}
+
+		return g.Run(".", gentronics.Data{})
+	},
+}
+
+func buildActionAppender(namespace, action string) gentronics.Runnable {
+	aa := actionAppender{namespace, action, inflect.Underscore(namespace)}
+	return aa
+}
+
+type actionAppender struct {
+	Namespace  string
+	ActionName string
+	FileName   string
+}
+
+func (aa actionAppender) Run(rootPath string, data gentronics.Data) error {
```

Comment:
I moved this to a `gentronics.Runnable` that will append to existing file, is that ok ? is there a gentronic default runnable that appends to a file ?